### ISSUE TITLE
Feature Request: Abstract Hyperparameters into YAML/JSON Configuration Files for Agentic Automation

### ISSUE THREAD

**Issue by abhiram123467** [ML4SCI/DeepLense#208]
As DeepLense moves towards integrating Agentic AI for autonomous gravitational lensing simulations, the current pipeline relies heavily on hardcoded hyperparameters and file paths within individual training scripts and notebooks. This makes it difficult for an autonomous agent (or an external orchestrator) to dynamically trigger training runs, modify parameters, or run hyperparameter sweeps without manually altering the Python code.

Describe the solution you'd like
I propose abstracting all model configurations (batch size, learning rate, epochs, dataset paths, random seeds) out of the training loops and into structured config.yaml or config.json files.

We can then implement a centralized configuration loader using Python's yaml or json libraries, combined with argparse, so that scripts can be executed headlessly by an AI agent like this:
python train_model.py --config configs/transformer_sim.yaml

Proposed Implementation Steps

Introduce a configs/ directory to store template YAML files for different architectures (e.g., Transformers, Diffusion models).

Write a utils/config_loader.py script to parse these files and feed them into the PyTorch/TensorFlow training loops.

Refactor one of the existing baseline training scripts to accept the config file rather than using hardcoded local variables.

Ensure backward compatibility by setting the config parser to fall back to default arguments if no YAML is provided.

**Comment by vignathi123-vi** [comment]
Hi, I’d like to work on this.
What I plan is to implement a centralized YAML-based config system with argparse support and backward compatibility.
I’ll start with refactoring one training pipeline and extend it across modules.
